src/uberComponents/DetailPage/index.tsx

```diff
@@ -0,0 +1,97 @@
+import { Bundle, Resource } from 'fhir/r4b';
+import { useEffect, useState } from 'react';
+import { useParams, Route, Routes, useLocation, Link, useNavigate, Params } from 'react-router-dom';
+
+import { SearchParams, RenderRemoteData, useService, WithId } from '@beda.software/fhir-react';
+
+import { PageContainer } from 'src/components';
+import { RouteItem } from 'src/components/BaseLayout/Sidebar/SidebarTop';
+import { Tabs } from 'src/components/Tabs';
+import { getFHIRResources } from 'src/services';
+import { compileAsFirst } from 'src/utils';
+
+export interface Tab<R extends Resource> {
+    label: string;
+    path?: string;
+    component: (resource: R) => JSX.Element;
+}
+
+interface DetailPageProps<R extends Resource> {
+    resourceType: R['resourceType'];
+    getSearchParams: (params: Readonly<Params<string>>) => SearchParams;
```

Comment:
Params<string> should be more specific because you're hard-coded to `id` below  
```
const menuItems: RouteItem[] = tabs.map(({ label, path }) => ({
        label,
        path: `/${basePath}/${params.id}/${path}`,
});
```


I suggest Params & {id: string}

I think something like